PSRabbitMq/Send-RabbitMqMessage.ps1

```diff
@@ -63,21 +102,51 @@
         [string]$Key,
 
         [Parameter(Mandatory=$true,ValueFromPipeline=$true)]
+        [Alias('Payload')]
         $InputObject,
 
         [switch]$Persistent,
 
         [Int32]$Depth = 2,
 
-        [PSCredential]$Credential,
+        [PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential,
 
         [System.Security.Authentication.SslProtocols]$Ssl,
 
-        [parameter(Mandatory = $false)]
-        [string]$vhost
+        [string]$vhost = '/',
+
+        [ValidateSet('application/clixml+xml','application/json','text/xml', 'text/plain')]
+        [string]$ContentType = 'application/clixml+xml',
+
+        [string]$ReplyTo,
+
+        [RabbitMQ.Client.PublicationAddress]$ReplyToAddress,
+
+        [string]$CorrelationID,
+
+        [switch]$Anonymous, 
+
+        [ValidateRange(0,9)]
+        [byte]$Priority,
+
+        [validateSet(1,2)]
+        [byte]$DeliveryMode,
+
+        [string]$Type,
+
+        [string]$MessageID,
+
+        [datetime]$timestamp,
+
+        [Int64]$TTL,
+
+        [System.Collections.Generic.Dictionary[String,Object]]$headers
```

Comment:
I left this like that because I did not have much need for it and that's what the dotNet class accept.
At least it can be used, but might need improvement.
Not sure I will have time to fix this properly, should I remove the parameter in the meantime?